src/main/scala/vsys/blockchain/contract/ContractVSwap.scala

```diff
@@ -75,29 +75,33 @@ object ContractVSwap {
     cdbvrGet ++ Array(tokenLiquidityIdStateVar.index, 5.toByte),
     basicConstantGet ++ DataEntry(Array(0.toByte), DataType.Boolean).bytes ++ Array(6.toByte),
     compareBytesEqual ++ Array(2.toByte, 3.toByte, 7.toByte),
+    basicConstantGet ++ DataEntry(Array(0.toByte), DataType.Boolean).bytes ++ Array(8.toByte),
     basicConstantGet ++ DataEntry(genFunctionOpcs(
       Seq(
+        assertEqual ++ Array(6.toByte, 8.toByte),
         cdbvMapValAdd ++ Array(tokenABalanceMap.index, 0.toByte, 1.toByte),
         basicConstantGet ++ DataEntry(Array(1.toByte), DataType.Boolean).bytes ++ Array(6.toByte),
       )
-    ), DataType.OpcBlock).bytes ++ Array(8.toByte),
-    compareBytesEqual ++ Array(2.toByte, 4.toByte, 9.toByte),
+    ), DataType.OpcBlock).bytes ++ Array(9.toByte),
+    compareBytesEqual ++ Array(2.toByte, 4.toByte, 10.toByte),
     basicConstantGet ++ DataEntry(genFunctionOpcs(
       Seq(
+        assertEqual ++ Array(6.toByte, 8.toByte),
         cdbvMapValAdd ++ Array(tokenBBalanceMap.index, 0.toByte, 1.toByte),
         basicConstantGet ++ DataEntry(Array(1.toByte), DataType.Boolean).bytes ++ Array(6.toByte),
       )
-    ), DataType.OpcBlock).bytes ++ Array(10.toByte),
-    compareBytesEqual ++ Array(2.toByte, 5.toByte, 11.toByte),
+    ), DataType.OpcBlock).bytes ++ Array(11.toByte),
+    compareBytesEqual ++ Array(2.toByte, 5.toByte, 12.toByte),
     basicConstantGet ++ DataEntry(genFunctionOpcs(
       Seq(
+        assertEqual ++ Array(6.toByte, 8.toByte),
         cdbvMapValAdd ++ Array(liquidityBalanceMap.index, 0.toByte, 1.toByte),
         basicConstantGet ++ DataEntry(Array(1.toByte), DataType.Boolean).bytes ++ Array(6.toByte),
       )
-    ), DataType.OpcBlock).bytes ++ Array(12.toByte),
-    conditionIf ++ Array(7.toByte, 8.toByte),
-    conditionIf ++ Array(9.toByte, 10.toByte),
-    conditionIf ++ Array(11.toByte, 12.toByte),
+    ), DataType.OpcBlock).bytes ++ Array(13.toByte),
+    conditionIf ++ Array(7.toByte, 9.toByte),
+    conditionIf ++ Array(10.toByte, 11.toByte),
+    conditionIf ++ Array(12.toByte, 13.toByte),
     assertTrue ++ Array(6.toByte)
   )
   lazy val depositTrigger: Array[Byte] = getFunctionBytes(depositId, onDepositTriggerType, nonReturnType, depositDataType, depositTriggerOpcs)
```

Comment:
So, why withdraw trigger need not add these constraints? @twwu123

Reply:
It shouldn't be a problem as users are unable to deposit duplicated token ids. The only possible function is to withdraw 0 tokens. This has been tested on the devnet, the token balances do not change.

I have also tested making liquidity token different, while token a and token b is the same. This results in me being able to deposit liquidity tokens but not token a / token b.